packages/create-block/lib/templates.js

```diff
@@ -98,9 +120,48 @@ const hasWPScriptsEnabled = ( templateName ) => {
 	return getTemplate( templateName ).wpScriptsEnabled || false;
 };
 
+const checkIsCoreTemplate = async templateName => templates[ templateName ] || false;
+
+const checkIsExternalTemplate = async templateName => {
+	try {
+		await command( `npm view ${ templateName }` );
+		return true;
+	} catch ( error ) {
+		return false;
+	}
+}
+
+const downloadExternalTemplate = async templateName => {
+	try {
+		const cwd = join( process.cwd(), 'temp' );
```

Comment:
You could extract it to the `getTempFolder` function to be able to reuse it later when cleaning up after the process ends.

Actually, you probably should also use the temp directory from your filesystem. This is how it's done for the script that releases the Gutenberg plugin:
https://github.com/WordPress/gutenberg/blob/c9ce32fb147b864d7f1d15df8ede97699ded4e91/bin/commander.js#L27

Reply:
Implemented that  👍